### PR TITLE
Fix non-deterministic map matrix dimension bug that causes bad Decode().

### PR DESCRIPTION
**Pull request by andychu**
- bin/test.sh: Change to h=1 to trigger the bug more reliably
- move validation of inputs to Decode(), instead of decode_dist.R, so we
  get it whenever we call Decode()
- Check dimensions in CreateAssocStringMap, for a better error message
- Require 'params' when calling ReadMapFile/LoadMapFile, so the bug can't happen in the first place
- Log a message when entries are removed from the map

### CODE REVIEW

File: analysis/R/read_input.R [hunk 1]
```diff
@@ -67,7 +67,7 @@ ReadCountsFile <- function(counts_file, params = NULL) {
   as.matrix(counts)
 }
 
-ReadMapFile <- function(map_file, params = NULL) {
```
Comment (tkaitchuck) [inline analysis/R/read_input.R]: The same problem exists in ReadCountsFile above.

File: bin/decode_dist.R [hunk 2]
```diff
@@ -83,34 +83,6 @@ AdjustCounts <- function(counts, params) {
   })
 }
 
-ValidateInput <- function(params, counts, map) {
```
Comment (tkaitchuck) [inline bin/decode_dist.R]: Why is this being removed?

Comment (andychu) [inline bin/decode_dist.R]: It is moved to Decode(), as CheckDecodeInputs.  The change description
mentions that.

Andy

**Comment by tkaitchuck** [comment]
I made the change to ReadCountsFile in the tkaitchuck branch. Either add it here, or we can merge twice.

**Comment by andychu** [comment]
Oh I see what you mean, I already merged this to master yesterday, because
it was getting in the way of actually fixing these bugs.

https://github.com/google/rappor/commit/44dcf81d57dfc8863ab25ee9bde0e1db8800c8a0

**Pull request merged**